application/codeaction/handlers.go

```diff
@@ -0,0 +1,77 @@
+package codeaction
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/rs/zerolog/log"
+
+	"github.com/snyk/snyk-ls/application/server/lsp"
+)
+
+const debounceDuration = 50 * time.Millisecond
+
+type TextDocumentCodeActionHandler func(context.Context, lsp.CodeActionParams) ([]lsp.CodeAction, error)
+type CodeActionResolveHandler func(context.Context, lsp.CodeAction) (*lsp.CodeAction, error)
+
+// ResolveCodeActionHandler returns a jrpc2.Handler that can be used to handle the "codeAction/resolve" LSP method
+func ResolveCodeActionHandler(service CodeActionsService) CodeActionResolveHandler {
+	logger := log.Logger.With().Str("method", "ResolveCodeActionHandler").Logger()
+
+	return func(ctx context.Context, params lsp.CodeAction) (*lsp.CodeAction, error) {
+		logger := logger.With().Interface("request", params).Logger()
+		logger.Info().Msg("RECEIVING")
+
+		action, err := service.ResolveCodeAction(params)
+		if err != nil {
+			logger.Error().Err(err).Msg("Failed to resolve code action")
+			return nil, err
+		}
+		logger.Info().Any("response", action).Msg("SENDING")
+		return &action, nil
+	}
+}
+
+// GetCodeActionHandler returns a jrpc2.Handler that can be used to handle the "textDocument/codeAction" LSP method
+func GetCodeActionHandler(service CodeActionsService) TextDocumentCodeActionHandler {
+	// We share a mutex between all the handlers to prevent concurrent runs.
```

Comment:
Not all the handlers, but all calls of getCodeAction :)

Reply:
Fixed